rq/queue.py

```diff
@@ -260,24 +260,26 @@ def enqueue(self, f, *args, **kwargs):
                                  description=description, depends_on=depends_on,
                                  job_id=job_id, at_front=at_front)
 
-    def enqueue_job(self, job, at_front=False):
+    def enqueue_job(self, job, pipeline=None, at_front=False):
         """Enqueues a job for delayed execution.
 
         If Queue is instantiated with async=False, job is executed immediately.
         """
-        with self.connection._pipeline() as pipeline:
-            # Add Queue key set
-            self.connection.sadd(self.redis_queues_keys, self.key)
-            job.set_status(JobStatus.QUEUED, pipeline=pipeline)
+        if not pipeline:
+            pipeline = self.connection._pipeline()
 
-            job.origin = self.name
-            job.enqueued_at = utcnow()
+        # Add Queue key set
+        self.connection.sadd(self.redis_queues_keys, self.key)
+        job.set_status(JobStatus.QUEUED, pipeline=pipeline)
 
-            if job.timeout is None:
-                job.timeout = self.DEFAULT_TIMEOUT
-            job.save(pipeline=pipeline)
+        job.origin = self.name
+        job.enqueued_at = utcnow()
 
-            pipeline.execute()
+        if job.timeout is None:
+            job.timeout = self.DEFAULT_TIMEOUT
+        job.save(pipeline=pipeline)
+
+        pipeline.execute()
```

Comment:
If `pipeline` is supplied as an argument, we shouldn't call `execute`, leaving the `execute` call to be called from the place where this method is called.